code/game/objects/items/hand_items.dm

```diff
@@ -285,6 +337,33 @@
 	blown_kiss.fire()
 	qdel(src)
 
+/obj/item/kisser/on_offered(mob/living/carbon/offerer)
+	. = TRUE
+
+	if(!(locate(/mob/living/carbon) in orange(1, offerer)))
+		return
+
+	offerer.visible_message(span_notice("[offerer] leans in slightly, offering a kiss on the cheek!"), \
+		span_notice("You lean in slightly, indicating you'd like to offer a kiss on the cheek!"), null, 2)
+	offerer.apply_status_effect(STATUS_EFFECT_OFFERING, src)
+
+/obj/item/kisser/on_offer_taken(mob/living/carbon/offerer, mob/living/carbon/taker)
+	. = TRUE
+
+	var/obj/projectile/blown_kiss = new kiss_type(get_turf(offerer))
+	offerer.visible_message("<b>[offerer]</b> gives [taker] \a [blown_kiss] on the cheek!", span_notice("You give [taker] \a [blown_kiss] on the cheek!"), ignored_mobs = taker)
+	to_chat(taker, span_nicegreen("[offerer] gives you \a [blown_kiss] on the cheek!"))
+	//Shooting Code:
+	blown_kiss.spread = 0
+	blown_kiss.original = taker
+	blown_kiss.fired_from = offerer
+	blown_kiss.firer = offerer // don't hit ourself that would be really annoying
+	blown_kiss.impacted = list(offerer = TRUE) // just to make sure we don't hit the wearer
+	blown_kiss.preparePixelProjectile(taker, offerer)
+	blown_kiss.suppressed = SUPPRESSED_VERY
+	blown_kiss.fire()
```

Comment:
This seems hacky if it's what I think it is, what's it for in this specific case? Is it for the very short effect of the heart blowing, or is it just for all the other effects?

Reply:
It's for all of it, not just for the effect of it being blown. I'd force it to directly impact the target instead of doing the firing, but i just know there'd be some edgecase it'd cause issues with projectiles not having their angles or something like that

Reply:
As long as is this isn't just an extremely complicated way to call the kiss's impact event, that's all I was worried about